src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/LifeCycle/LifeCycleEffectRouter.uwp.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Linq;
+using Windows.UI.Xaml;
+using Xamarin.CommunityToolkit.Effects;
+using Xamarin.CommunityToolkit.UWP.Effects;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.UWP;
+
+[assembly: ExportEffect(typeof(LifeCycleEffectRouter), nameof(LifeCycleEffect))]
+
+namespace Xamarin.CommunityToolkit.UWP.Effects
+{
+	public class LifeCycleEffectRouter : PlatformEffect
+	{
+		FrameworkElement nativeView;
+		LifeCycleEffect lifeCycleEffect;
+
+		protected override void OnAttached()
+		{
+			lifeCycleEffect = Element.Effects.OfType<LifeCycleEffect>().FirstOrDefault();
+
+			_ = lifeCycleEffect ?? throw new ArgumentNullException($"The effect {nameof(LifeCycleEffect)} can't be null.");
+
+			nativeView = Control ?? Container;
+
+			nativeView.Loaded += OnNativeViewLoaded;
+			nativeView.Unloaded += OnNativeViewUnloaded;
+		}
+
+		void OnNativeViewLoaded(object sender, RoutedEventArgs e) => lifeCycleEffect.RaiseLoadedEvent(Element);
+
+		void OnNativeViewUnloaded(object sender, RoutedEventArgs e)
+		{
+			lifeCycleEffect.RaiseUnloadedEvent(Element);
+			nativeView.Unloaded -= OnNativeViewUnloaded;
+			nativeView.Loaded -= OnNativeViewLoaded;
+			lifeCycleEffect = null;
+			nativeView = null;
```

Comment:
Not really, because if you leave the screen the `OnDeatached` method isn't called after the `Unloaded` event, in other words, the Effect will be detached before the control is unloaded and we want to make sure that doesn't happen. And if the control is Unloaded, the Effect will be detached already.

I'm still thinking about how to make this more _beaultiful_, maybe an external method and calling on both the `OnDeatached` method and the `Unloaded` event.

Reply:
Got it.
I'm just thinking about the case when you add effect programmatically and then remove it.